sirepo/package_data/static/js/raydata.js

```diff
@@ -637,25 +692,48 @@ SIREPO.app.directive('scanSelector', function() {
-                <thead>
-                  <tr>
-                    <th data-ng-repeat="h in getHeader()">{{ h }}</th>
-                  </tr>
-                </thead>
-                <tbody ng-repeat="s in scans">
-                  <tr>
-                    <td><input type="checkbox" data-ng-checked="s.selected" data-ng-click="toggleScanSelection(s)"/></td>
-                    <td data-ng-repeat="c in getHeader().slice(1)">{{ getScanField(s, c) }}</td>
+              <div data-ng-if="searchStartTime && searchStopTime">
+	        <button class="btn btn-info btn-xs" data-ng-click="newCommand()" style="float: right;"><span class="glyphicon glyphicon-plus"></span></button>
+                <table class="table table-striped table-hover col-sm-4">
+                  <thead>
+                    <tr>
+                      <th data-ng-repeat="column in columns track by $index" data-ng-mouseover="hoverChange($index, true)" data-ng-mouseleave="hoverChange($index, false)" style="width: 100px; height: 40px;">
+                            {{ column }}
+                            <button type="submit" class="btn btn-primary btn-xs"  id="del-column" data-ng-show="showDeleteButton($index)" data-ng-click="deleteCol(column)"><span class="glyphicon glyphicon-remove"></span></button>
+                      </th>
                   </tr>
-                </tbody>
-              </table>
+                  </thead>
+                  <tbody ng-repeat="s in scans">
+                    <tr>
+                      <td style="width: 100px; height: 40px;"><input type="checkbox" data-ng-checked="s.selected" data-ng-click="toggleScanSelection(s)"/></td>
+                      <td data-ng-repeat="c in columns.slice(1)" style="width: 100px; height: 40px;">{{ getScanField(s, c) }}</td>
+                    </tr>
+                  </tbody>
+                </table>
+              </div>
 	    </div>
+
+	    <div data-column-picker="" data-title="Add Column" data-id="sr-columnPicker-editor" data-available-columns="availableColumns" data-master-list-columns="masterListColumns" data-columns="columns" data-default-columns="defaultColumns" data-set-columns="setColumns" data-set-available-columns="setAvailableColumns"></div>
+
         `,
-        controller: function(appState, errorService, panelState, raydataService, requestSender, timeService, $scope) {
+        controller: function(appState, errorService, panelState, raydataService, requestSender, timeService, $scope, $interval) {
+	    let hoveredIndex = null;
 	    let cols = [];
 	    const startOrStop = ['Start', 'Stop'];
+
 	    $scope.scans = [];
+	    $scope.defaultColumns = ['start', 'stop', 'suid'];
+	    $scope.columns = ['select'].concat($scope.defaultColumns.concat(appState.models.metadataColumns.selected));
+	    $scope.masterListColumns = [];
+	    $scope.availableColumns = [];
+
+	    $scope.deleteCol = function(colName) {
+		let index = appState.models.metadataColumns.selected.indexOf(colName);
```

Comment:
const for `index`. Also, no need fo a temp variable can use
`appState.models.metadataColumns.selected.splice(
    appState.models.metadataColumns.selected.indexOf(colName),
    1
);`

Reply:
e64969d8a3a9303ebcf97566971db0411515341f